### DISCUSSION TITLE
Very low FPS (SSO selection)

### DISCUSSION THREAD

**Discussion by StarFlea** [Stellarium/stellarium#3265] [Q&A] [answered]
Hello!

In the current version 23.1, my FPS (of around 13 FPS) collapse as soon as I select a **solar system object**.
The worst is when selecting the moon (sometimes < 1 FPS).
DSO selection is not affected.

Just zooming in on the moon without selecting it doesn't cause any change in FPS either.

And if I cancel the selection (right mouse button), then the FPS increased again.

Actually, my computer is relatively well equipped:
CPU: Intel Core i7-10750H (Six Core)
RAM: 64GB
GPU: NVIDIA GeForce GTX 1660 Ti (with latest driver)
Win10

As a test, I switched off the 3D models, but without any effect...

Is this a fundamental problem or is there a setting in the config.ini that could lead to success?

StarFlea

**Comment by 10110111** [comment]
It's not normal, something is wrong with your configuration. Do you choose the correct shortcut (not Mesa and not ANGLE)?

**Comment by StarFlea** [comment]
Yes, I start in standard mode.

But just for info: In ANGLE mode or MESA mode, the FPS also decrease when I select a SSO.

**Reply by gzotti** [reply]
FPS decreases when selecting SSO and providing data output, because some detail computation takes its time. However, 13->1fps is certainly not OK. 
You can maybe play with the Information settings (configuration dialog, Information tab) and find out which data output setting is the offender. But again, on a 10th-gen i7 this should not happen. The logfile may contain more info.

**Comment by StarFlea** [comment, marked as answer]
**Thanks, that was the right hint! :)**

The main responsible seems to be the "rises, transits and sets" checkbox. Here the FPS increase massively when I disable them (from <1 to 8).

In addition, the "proper motion" and "Additional information" checkboxes have greater effects...

Now "rises, transits and sets" would be useful information (I can also do without the others personally).
Since the FPS are permanently decreased, does this information seem to be calculated continuously?

**Reply by gzotti** [reply]
The RTS calculation is pretty involved, esp. for the Moon. Are you living e.g. in high latitudes? This may explain a longer iteration count to get the right result.

**Reply by StarFlea** [reply]
Well, I wouldn't necessarily say high latitude with N 48° 37'...

If the FPS would drop for a few moments, that would be ok, but RTS keeps it extremely low...

**Reply by gzotti** [reply]
OK, similar latitude. My fps drops from 57 to 16 for the Moon with "all available" information and 22fps without RTS. So yes, it is costly. Maybe we could cache the result, but find a good criterium for recomputation (just date change? Actually this showed earlier not to be enough.)

**Reply by StarFlea** [reply]
A good question!

Just a few quick ideas:
A recalculation would also be necessary for a change of location.
At the beginning one could calculate RTS and trigger a recalculation only when a new period outside of this rise or set is reached...?

Is RTS calculated iteratively?
Maybe there are still ways to optimize the contained inner loops of the algorithm?

On the other hand, I still don't quite understand why the FPS are decreasing. After all, RTS is calculated using the CPU, right?
But for rendering the graphical display is the GPU used?

Just guesses on my part...
